app/helpers/repository_helper.rb

```diff
@@ -136,6 +136,13 @@ def add_folder(folder_path, user, repo, path: '/', txn: nil, required_files: nil
     # check if only required files are allowed for a submission
     # allowed folders = paths in required files
     if required_files.present? && required_files.none? { |file| file.starts_with?(folder_path) }
+      folder_path = format_folder_path folder_path
```

Comment:
What's going on with `format_folder_path`? What is it removing?

Reply:
Thanks David, as an error message it would display the pathname variable which includes the internal storage directories. e.g. for assignment 'A0', it will include it in the pathname "/A0/subdir/...". We prefer to hide that information from the end user. That is what format_folder_path does.